Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -115,3 +116,36 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
         }
     }
 }
+
+enum BuildSettingsError: FatalError {
+    case invalidValue(Any)
+
+    var description: String {
+        switch self {
+        case let .invalidValue(value):
+            return "Cannot convert \"\(value)\" to SettingValue type"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .invalidValue:
+            return .bug
+        }
+    }
+}
+
+extension BuildSettings {
+    func toSettings() throws -> [String: SettingValue] {
+        return try mapValues { value in
+            switch value {
+            case let value as String:
+                return .string(value)
+            case let value as [String]:
+                return .array(value)
+            default:
+                throw BuildSettingsError.invalidValue(value)
```

Comment:
Can we prevent not having to throw here by leveraging types?

Reply:
@pepibumur not sure. The default case is something we do not expect (and should never happen - i.e. invalid default data from the `XcodeProj.BuildSettingsProvider.swift`). it's probably better to throw and stop processing than quietly assign something like `.unknown` and continue generation.

Can you see any nice alternatives?

Reply:
We can go ahead with this. If we see this happening for some reason, we can think of a different way to handle this.